src/rust/ensogl/lib/components/src/toggle_button.rs

```diff
@@ -202,26 +205,30 @@ impl<Shape:ColorableShape+'static> ToggleButton<Shape>{
 
              // === Input Processing ===
 
+            toggle <- any(...);
+            toggle <+ frp.toggle;
             eval frp.set_size ((size) model.icon.size().set(*size));
+            frp.source.state <+ toggle.toggle();
+            frp.source.state <+ frp.set_state;
 
 
             // === Mouse Interactions ===
 
             frp.source.mouse_over <+ icon.mouse_over;
             frp.source.mouse_out  <+ icon.mouse_out;
-            frp.source.state      <+ icon.mouse_down.toggle();
+            toggle                <+ icon.mouse_down;
 
 
             // === Color ===
 
             invisible <- frp.set_visibility.on_false().constant(0.0);
             color.target_alpha <+ invisible;
 
-            visible    <- frp.set_visibility.gate(&frp.set_visibility);
-            is_hovered <- bool(&icon.mouse_out,&icon.mouse_over);
-            is_pressed <- bool(&icon.mouse_up,&icon.mouse_down);
+            frp.source.visible <+ frp.set_visibility;
+            is_hovered         <- bool(&icon.mouse_out,&icon.mouse_over);
+            is_pressed         <- bool(&icon.mouse_up,&icon.mouse_down);
```

Comment:
`is_hovered` and `is_pressed` should be exposed from icon directly as this is very common thing to check